### PR TITLE
Add travis test for DockerHub images

### LINKED ISSUE #11: Travis should test pulling external images for build
Currently its following the complete build process, including building images locally

### PR DESCRIPTION
**Pull request by Tomcli**
- Add a new Travis job to test DockerHub images and modify Travis to run 2 test jobs in parallel.
- Move the test script commands under `etc/travis`.
- Add new helper make function `pull-dockerhub-images` to pull the DockerHub images with tag `0.0.1-master`.
- Fix #11

### CODE REVIEW

**Comment by animeshsingh** [comment]
Thanks @Tomcli 
Can we give the current images a version number instead of pulling latest?

File: Makefile [hunk 1]
```diff
@@ -203,8 +207,27 @@ test-submit:      ## Submit test training job
 				kubectl get pods | grep learner- | awk '{print $$1}' | xargs -I '{}' kubectl logs '{}' -c load-data; \
 				exit 1);
 
+pull-dockerhub-images:  ## Pull FfDL images from dockerhub
+pull-dockerhub-images: $(addprefix pull-, $(TEST_IMAGES))
+
+$(addprefix pull-, $(TEST_IMAGES)): pull-%: %
+	@TRAVIS_IMAGES=$< make .pull-dockerhub-images
```
Comment (whummer) [inline Makefile]: nitpick: perhaps better `TRAVIS_IMAGE` instead of `TRAVIS_IMAGES`? (also in lines 221, 226, 229)

File: Makefile [hunk 2]
```diff
@@ -22,6 +22,10 @@ UI_REPO = git@github.com:IBM/FfDL-dashboard.git
 CLI_CMD = $(shell pwd)/cli/bin/ffdl-$(UNAME_SHORT)
 CLUSTER_NAME ?= mycluster
 PUBLIC_IP ?= 127.0.0.1
+TRAVIS_IMAGES_VERSION ?= 0.0.1-master
+TEST_IMAGES = $(addprefix $(DOCKER_NAMESPACE)/, $(TEST_IMAGES_SUBFIX))
```
Comment (whummer) [inline Makefile]: nitpick: I think "suffix" would be the more common terminology (`TEST_IMAGES_SUFFIX`)

File: etc/travis/test-images-dockerhub.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# fail fast
```
Comment (whummer) [inline etc/travis/test-images-dockerhub.sh]: Given how we're calling these scripts from `.travis.yml`, do we need to add [shebangs](https://en.wikipedia.org/wiki/Shebang_(Unix)) at the top of these scripts?

```
#!/bin/bash
...
```

File: etc/travis/test-images-dockerhub.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+set -e
+export MAKE_ARGS=--no-print-directory
+# pull images from dockerhub
+make $MAKE_ARGS pull-dockerhub-images
+make $MAKE_ARGS tag-dockerhub-images-to-latest
```
Comment (whummer) [inline etc/travis/test-images-dockerhub.sh]: Looks like a good workaround solution for now, but I guess at some point we should make the Docker tags parameterizable in the deployment script, so we don't need to overwrite `latest` tag here.

**Comment by Tomcli** [comment]
Moved to #111

**Pull request closed without merging**